JASP-Engine/JASP/R/multinomialtest.R

```diff
@@ -15,546 +15,447 @@
-    plotFrame$yAxisMargin <- plotFrame$upperCI
-    for(i in 1:nrow(plotFrame)){
-      if(plotFrame$upperCI[i] == 0){
-        plotFrame$yAxisMargin[i] <- plotFrame$obs[i]
-      }   
-    }
-
-    # Create plot
-    p <- ggplot2::ggplot(data = plotFrame,
-                         mapping = ggplot2::aes(x = factor, y = obs)) +
-      ggplot2::geom_bar(stat = "identity", size = 0.75, colour="black",
-                        fill = "grey") +
-      ggplot2::geom_errorbar(ggplot2::aes(ymin = ciDf$lowerCI,
-                                          ymax = ciDf$upperCI),
-                             size = 0.75, width = 0.3) +
-      base_breaks_y(plotFrame$yAxisMargin) +
-      ggplot2::xlab(options$factor) +
-      ggplot2::ylab(yname)
-
-    p <- JASPgraphs::themeJasp(p, horizontal = TRUE, xAxis = FALSE)
-
-
-
-    # create plot object
-    content <- .writeImage(width = options$plotWidth,
-                           height = options$plotHeight,
-                           plot = p, obj = TRUE)
-
-    descriptivesPlot[["convertible"]] <- TRUE
-    descriptivesPlot[["obj"]] <- content[["obj"]]
-    descriptivesPlot[["data"]] <- content[["png"]]
-    descriptivesPlot[["status"]] <- "complete"
-
+.multinomialDescriptivesPlotFill <- function(jaspResults, descriptivesPlot, options) {
+  # Get Results
+  chisqResults <- jaspResults[["stateChisqResults"]]$object
+  # Generate the plot
+  f <- names(chisqResults[[1]][["observed"]])
+  plotFrame <- data.frame(factor = factor(f, levels = rev(f)))
+  # Counts or props
+  if (options$countProp == "descCounts") {
+    yname <- "Observed counts"
+    obs   <- as.integer(chisqResults[[1]][["observed"]])
   } else {
-    descriptivesPlot[["data"]] <- ""
+    div   <- sum(chisqResults[[1]][["observed"]])
+    yname <- "Observed proportions"
+    obs   <- as.numeric(chisqResults[[1]][["observed"]])/div
   }
-
-  return(descriptivesPlot)
+  plotFrame <- cbind(plotFrame, obs)
+  
+  # Calculate confidence interval
+  if (options$descriptivesPlotConfidenceInterval){
+    ciDf       <- .multComputeCIs(chisqResults[[1]][["observed"]], 
+                                  options$descriptivesPlotConfidenceInterval, 
+                                  ifErrorReturn = 0, scale = options$countProp)
+    plotFrame  <- cbind(plotFrame, ciDf)
+  }
+  
+  # Define custom y axis function
+  base_breaks_y <- function(x){
+    b <- pretty(c(0,x))
+    d <- data.frame(x = -Inf, xend = -Inf, y = min(b), yend = max(b))
+    list(ggplot2::geom_segment(data = d, 
+                               ggplot2::aes(x = x, y = y, xend = xend, yend = yend),
+                               size = 0.75, inherit.aes = FALSE),
+         ggplot2::scale_y_continuous(breaks = b))
+  }
+  
+  # Determine y-axis margin: If CIs could not be computed, use observed counts
+  plotFrame$yAxisMargin <- plotFrame$upperCI
+  for(i in 1:nrow(plotFrame))
+    if(plotFrame$upperCI[i] == 0)
```

Comment:
I can't judge if this will ever be exactly 0. If upperCI is a float, it could be better to test against a tolerance (e.g., `abs(plotFrame$upperCI) <= sqrt(.Machine$double.eps)`)